makefiles/docker.inc.mk

```
@@ -121,12 +121,13 @@ DOCKER ?= docker
 _docker_is_podman = $(shell $(DOCKER) --version | grep podman 2>/dev/null)
 
 # Set default run flags:
-# - allocate a pseudo-tty
+# - always pull to have riot/riotbuild:latest up-to-date
 # - remove container on exit
+# - allocate a pseudo-tty
 # - set username/UID to executor
 DOCKER_USER ?= $$(id -u)
 DOCKER_USER_OPT = $(if $(_docker_is_podman),--userns keep-id,--user $(DOCKER_USER))
-DOCKER_RUN_FLAGS ?= --rm --tty $(DOCKER_USER_OPT)
+DOCKER_RUN_FLAGS ?= --pull=always --rm --tty $(DOCKER_USER_OPT)
```

```suggestion
DOCKER_RUN_FLAGS ?= --pull=newer --rm --tty $(DOCKER_USER_OPT)
```

With `podman` this does the trick. It does take a bit of time to timeout the check in the Brandenburg case.

But maybe the best would be to just allow users to overwrite this via environment variable, e.g. by setting `RIOT_DOCKER_PULL_POLICY` or whatever.

---

Unfortunately this flag is not available for docker: https://docs.docker.com/reference/cli/docker/container/run/#pull

I've tried adding it and it will just be silently ignored (i.e., the docker image is not updated).

---

So we could have different defaults for docker (`--pull=always`) and podman (`--pull=newer`) and have `DOCKER_SKIP_PULL` to disable pulling altogether?